infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -410,7 +412,7 @@ private boolean isValidSnapshotStart(LogReplicationEntryMsg entry) {
      *
      * @param entry a SNAPSHOT_START message
      */
-    private void processSnapshotStart(LogReplicationEntryMsg entry) {
+    private synchronized void processSnapshotStart(LogReplicationEntryMsg entry) {
```

Comment:
What was the conflicting state here? snapshotSinkBufferManager is used at other places as well. For example: Receive and processReceivedMessage. Wondering if multiple threads come in sequentially updating the snapshotSinkBufferManager here, can cause conflict within other methods?

Reply:
yes, there could be...
my next task involves going into these classes more exhaustively and make the constructs thread safe. This bit here is a quick fix for now.

Reply:
what is the shared state here?  Each session has a dedicated SinkManager so not sure why the synchronization is required..

Reply:
The shared state identified as of yet is `rxState`. This gets updated by 2 threads within a session...by `LogReplicationServer-*` and `snapshotSyncApplyExecutor` thread